Python/ceval.c

```diff
@@ -2782,7 +2792,14 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, InterpreterFrame *frame, int thr
             traceback = exc_info->exc_traceback;
             exc_info->exc_type = POP();
             exc_info->exc_value = POP();
-            exc_info->exc_traceback = POP();
+
+            assert(exc_info->exc_value);
+            if (exc_info->exc_value != Py_None) {
+                exc_info->exc_traceback = PyException_GetTraceback(exc_info->exc_value);
```

Comment:
The next step is to remove exc_traceback from exc_info and then this will not be here.